apps/emqx_machine/src/emqx_cluster_rpc.erl

```diff
@@ -275,24 +306,71 @@ trans_query(TnxId) ->
             #{tnx_id => TnxId, mfa => MFA, initiator => InitNode, created_at => CreatedAt}
     end.
 
-apply_mfa(TnxId, {M, F, A} = MFA) ->
-    try
-        Res = erlang:apply(M, F, A),
-        Succeed =
-        case Res of
-             ok ->
-                 ?SLOG(notice, #{msg => "succeeded to apply MFA", tnx_id => TnxId, mfa => MFA, result => Res}),
-                 true;
-            {ok, _} ->
-                ?SLOG(notice, #{msg => "succeeded to apply MFA", tnx_id => TnxId, mfa => MFA, result => Res}),
-                true;
-            _ ->
-                ?SLOG(error, #{msg => "failed to apply MFA", tnx_id => TnxId, mfa => MFA, result => Res}),
-                false
+-define(TO_BIN(_B_), iolist_to_binary(io_lib:format("~p", [_B_]))).
+
+apply_mfa(TnxId, {M, F, A}) ->
+    Res =
+        try erlang:apply(M, F, A)
+        catch
+            C : E -> {crash, C, E}
         end,
-        {Succeed, Res}
-    catch
-        C : E ->
-            ?SLOG(critical, #{msg => "crash to apply MFA", tnx_id => TnxId, mfa => MFA, exception => C, reason => E}),
-            {false, lists:flatten(io_lib:format("TnxId(~p) apply MFA(~p) crash", [TnxId, MFA]))}
+    Meta = #{tnx_id => TnxId, module => M, function => F, args => ?TO_BIN(A)},
+    log_and_alarm(Res, Meta),
+    Succeed = (Res =:= ok orelse (is_tuple(Res) andalso Res =/= {} andalso element(1, Res) =:= ok)),
```

Comment:
the `ok` or `{ok, _}` check are duplicated in these two lines.
maybe create a `is_success/1` help function, to check the result, then call log_and_alarm before returning the result.